get-image-version/action.yml

```
@@ -0,0 +1,40 @@
+name: 'Get Image Version'
+description: 'Parses the version.json content'
+inputs:
+  IMAGE_CONTEXT_DIR:
+    description: 'Context for the version.json'
+    required: true
+outputs:
+  VERSION:
+    description: "Version; i.e.: 1.0.1"
+    value: ${{ steps.check_version.outputs.VERSION }}
+  MINOR_VERSION:
+    description: "Minor version; i.e.: 1.0"
+    value: ${{ steps.check_version.outputs.MINOR_VERSION }}
+  IMAGE_TAGS:
+    description: "Image tags from versions; i.e.: latest, 1.0.1 and 1.0"
+    value: ${{ steps.check_version.outputs.IMAGE_TAGS }}
+runs:
+  using: "composite"
+  steps:
+    - name: Check and verify version.json
+      id: check_version
+      shell: bash
+      run: |
+        VERSION=$(jq -r '.version' ${{ inputs.IMAGE_CONTEXT_DIR }}/version.json)
+
+        # version.json must have semantic version: vMAJOR.MINOR.PATCH
+        if [[ ! $VERSION =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+            echo "${{ inputs.IMAGE_CONTEXT_DIR }}/version.json does not contain semantic version, vMAJOR.MINOR.PATCH"
+            cat ${{ inputs.IMAGE_CONTEXT_DIR }}/version.json
+            exit 1
+        fi
```

Could we expand the regex to also accomodate for [semver pre-release versions](https://semver.org/#spec-item-9)? This would allow a version to be `v1.2.3-1` where the last digit can be incremented for changes other than the application version (e.g. Dockerfile changes) for 3rd party applications where we don't control the application version. 
```suggestion
        # version.json must have semantic version: vMAJOR.MINOR.PATCH(-PRE-RELEASE)
        if [[ ! $VERSION =~ ^v[0-9]+\.[0-9]+\.[0-9]+([0-9A-Za-z-]*)$ ]]; then
            echo "${{ inputs.IMAGE_CONTEXT_DIR }}/version.json does not contain semantic version, vMAJOR.MINOR.PATCH"
            cat ${{ inputs.IMAGE_CONTEXT_DIR }}/version.json
            exit 1
        fi
```